### PR TITLE
cppcheck: output all annotations as error

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
`cppcheck` has a non-zero exit code even on warnings, so to not confuse contributors, align the annotation output with that.

### Testing procedure
I can add a test commit if necessary.

### Issues/PRs references
None

### CODE REVIEW

**Review by bergzand — APPROVED** [review]
> I can add a test commit if necessary.

I believe you.

**Comment by miri64** [comment]
> I believe you.

I changed it now that it is only treating all output as error, when the exit code is non-zero :sweat_smile: Do you want another look?

File: dist/tools/cppcheck/check.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ if github_annotate_is_on; then
         DETAILS="${DETAILS}     * (reason: <your reason why you think this is "
         DETAILS="${DETAILS}a false positive>) */\n"
 
-        if echo "${SEVERITY}" | grep -q "\<error\>"; then
+        if [ ${RESULT} -ne 0 ] || echo "${SEVERITY}" | grep -q "\<error\>"; then
```
Comment (bergzand) [inline dist/tools/cppcheck/check.sh:83]: Mind adding a comment above this line to explain to the next reader why we always treat everything as error on non-zero exit code?

Comment (miri64) [reply dist/tools/cppcheck/check.sh:86]: Done and squashed.

**Review by bergzand — APPROVED** [review]
re-ack

**Pull request merged**